### ISSUE TITLE
Keep record on client if deleted on server (table is download-only sync) [SQLServer-SQLServer]

### ISSUE THREAD

**Issue by REMjn832** [Mimetis/Dotmim.Sync#365]
First, thank you for this project. I am using it now to sync 2 SQL Server databases. I need some tables to come from the server to the client (download-only), and other tables to go from the client to the server (upload-only). That part is working well using the syncConfig to set the direction of each table.

My question, is that for the download (from server to client), I don't actually want to delete records that have been deleted on the server. Right now, if I add or delete a record on the server, it will sync that change to the client - and in my case, I would like to keep the record on the client, even if the server deleted the record. Is that possible?

Thanks very much!
-Josh

**Comment by Mimetis** [comment]
Well, you can have the lines that "are going to be" deleted, on the client, using a `On...` methods (we call them `interceptors` )

```
agent.LocalOrchestrator.OnTableChangesApplying(tableArgs =>
{
 if (tableArgs.State == DataRowState.Deleted)
  {
    // get deleted lines
    var deleted = tableArgs.Changes;
   }
});

```

but unfortunatelly, this interceptor won't prevent the table deletion
It should not be too complicated to add a `tableArgs.AcceptApplyingChanges = [false / true]` 

I think you raised a very interesting point here.
I will for sure add this option in an upcoming release.

For now, there is no solution, but I will update the current release in less than 2 weeks, so expect it, before end of year.

**Comment by REMjn832** [comment]
This is great! I can wait. Thanks very much!

**Comment by Mimetis** [comment]
I will keep you updated on this :)

**Comment by Mimetis** [comment]
@gentledepp You could be interested in this one

**Comment by Mimetis** [comment]
@REMjn832 This feature will be available on the next relase `v0.6`
You can have a preview on this feature here : https://github.com/Mimetis/Dotmim.Sync/discussions/391#discussioncomment-267244
